src/main/java/pl/javastart/task/Main.java

```diff
@@ -1,16 +1,94 @@
 package pl.javastart.task;
 
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class Main {
+    private static final Pattern[] patterns = new Pattern[]{
+            Pattern.compile("(\\d{4})-(0[1-9]|1[0-2])-(\\d{2}) (\\d{2}):(\\d{2}):(\\d{2})"),
+            Pattern.compile("(\\d{2}).(0[1-9]|1[0-2]).(\\d{4}) (\\d{2}):(\\d{2}):(\\d{2})"),
+            Pattern.compile("(\\d{4})-(0[1-9]|1[0-2])-(\\d{2}$)")
+    };
 
     public static void main(String[] args) {
         Main main = new Main();
         main.run(new Scanner(System.in));
     }
 
-    public void run(Scanner scanner) {
-        // uzupełnij rozwiązanie. Korzystaj z przekazanego w parametrze scannera
+    public boolean validationWithRegex(String expression) {
+        boolean patternFind = false;
+
+        for (Pattern item : patterns) {
+            Matcher matcher = item.matcher(expression);
+            if (matcher.find()) {
+                patternFind = true;
+                break;
+            }
+        }
+        return patternFind;
+    }
+
+    public static String reformatData(String data) {
+        if (patterns[2].matcher(data).find()) {
+            return data + " 00:00:00";
+        } else if (patterns[1].matcher(data).find()) {
+            String[] parsedData = data.split("[. ]");
+            return parsedData[2] + "-" + parsedData[1] + "-" + parsedData[0] + " " + parsedData[3];
+        } else {
+            return data;
+        }
+    }
```

Comment:
To jest mało rozwojowe rozwiązanie. Dodając kolejne formaty, konieczne byłoby dodawanie tutaj coraz więcej ifów. Gdybyś użyła np. DateTimeFormattera, to można wtedy użyć pętli i dodawanie kolejnych formatów nie powinno wpływać na resztę kodu